src/topic/models.py

```diff
@@ -0,0 +1,244 @@
+    openalex_id = models.CharField(
+        unique=True,
+        blank=False,
+        null=False,
+        max_length=255,
+    )
+
+    domain = models.ForeignKey(
+        "topic.Domain",
+        related_name="fields",
+        blank=True,
+        null=True,
+        on_delete=models.CASCADE,
+    )
+
+    display_name = models.TextField(
+        blank=False,
+        null=False,
+    )
+
+
+class Subfield(DefaultModel):
+    # https://docs.openalex.org/api-entities/topics/topic-object#subfield
+    openalex_id = models.CharField(
+        unique=True,
+        blank=False,
+        null=False,
+        max_length=255,
+    )
+
+    field = models.ForeignKey(
+        "topic.Field",
+        related_name="subfields",
+        blank=True,
+        null=True,
+        on_delete=models.CASCADE,
+    )
+
+    display_name = models.TextField(
+        blank=False,
+        null=False,
+    )
+
+
+class Topic(DefaultModel):
+    # https://docs.openalex.org/api-entities/topics/topic-object#id
+    openalex_id = models.CharField(
+        unique=True,
+        blank=False,
+        null=False,
+        max_length=255,
+    )
+
+    # https://docs.openalex.org/api-entities/topics/topic-object#display_name
+    display_name = models.TextField(
+        blank=False,
+        null=False,
+    )
+
+    subfield = models.ForeignKey(
+        "topic.Subfield",
+        related_name="topics",
+        blank=True,
+        null=True,
+        on_delete=models.CASCADE,
+    )
+
+    # https://docs.openalex.org/api-entities/topics/topic-object#works_count
+    works_count = models.IntegerField(
+        blank=True,
+        null=True,
+        default=0,
+    )
+
+    # https://docs.openalex.org/api-entities/topics/topic-object#cited_by_count
+    cited_by_count = models.IntegerField(
+        blank=True,
+        null=True,
+        default=0,
+    )
+
+    # https://docs.openalex.org/api-entities/topics/topic-object#keywords
+    keywords = ArrayField(
+        models.CharField(max_length=255), blank=True, null=True, default=list
+    )
+
+    # https://docs.openalex.org/api-entities/topics/topic-object#created_date
+    openalex_created_date = models.DateTimeField(
+        blank=True,
+        null=True,
+    )
+
+    # https://docs.openalex.org/api-entities/topics/topic-object#updated_date
+    openalex_updated_date = models.DateTimeField(
+        blank=True,
+        null=True,
+    )
+
+    def upsert_from_openalex(oa_topic):
+        has_dates = oa_topic.get("updated_date") and oa_topic.get("created_date")
+
+        # Normalize created, updated dates to format that is compatible with django
+        oa_topic = OpenAlex.normalize_dates(oa_topic)
+
+        topic = None
+        try:
+            topic = Topic.objects.get(openalex_id=oa_topic["id"])
+        except Topic.DoesNotExist:
+            pass
+
+        # if topic exists, determine if we need to update it
+        needs_update = False
+        if topic and has_dates:
```

Comment:
Why might a topic not have dates? Are there scenarios where we would want to update topics that don't have dates?